File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        i=0
        j=n-1
        total=0
        while(i<=j):
            l=min(height[i],height[j])
            b=j-i
            total=max(total,l*b)
            if(height[i]<height[j]):
                i+=1
            else:
                j-=1
        
        return total

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int n=nums.size();
        int i=0,j=n-1,total=0;
        int l=0,b=0;
        while(i<=j){
            l=min(nums[i],nums[j]);
            b=j-i;
            total=max(total,l*b);
            if(nums[i]<nums[j])
            i++;
            else
            j--;
        }
        return total;
    }
};